pkg/gateway/model_build_lattice_service.go

```diff
@@ -132,28 +131,65 @@ func (t *latticeServiceModelBuildTask) buildLatticeService(ctx context.Context)
 		spec.CustomerDomainName = ""
 	}
 
-	if t.route.DeletionTimestamp().IsZero() {
-		spec.IsDeleted = false
-	} else {
-		spec.IsDeleted = true
+	certArn, err := t.getACMCertArn(ctx)
+	if err != nil {
+		return nil, err
+	}
+	spec.CustomerCertARN = certArn
+
+	svc, err := model.NewLatticeService(t.stack, spec)
+	if err != nil {
+		return nil, err
 	}
 
-	serviceResourceName := fmt.Sprintf("%s-%s", t.route.Name(), t.route.Namespace())
+	t.log.Debugf("Added service %s to the stack (ID %s)", svc.Spec.LatticeServiceName(), svc.ID())
+	svc.IsDeleted = !t.route.DeletionTimestamp().IsZero()
+	return svc, nil
+}
 
-	t.latticeService = model.NewLatticeService(t.stack, serviceResourceName, spec)
+// returns empty string if not found
+func (t *latticeServiceModelBuildTask) getACMCertArn(ctx context.Context) (string, error) {
```

Comment:
Previously this logic was embedded in the [build listener logic](https://github.com/aws/aws-application-networking-k8s/pull/457/files#diff-bcc1d433a3e50c0ffedc6f1ea18fef5a9fdc2e22aa2331c03037b9627dd0d31bL91) as part of ```extractListenerInfo```, though it would reach back and update the service model. I've just moved the logic to where I think it should more naturally live.